### PR TITLE
36 fix test runner

### PR DESCRIPTION
**Pull request by LukeParky**
DESCRIPTION OF PR:
Makes it so unit tests can run on the github actions runner. None of the skipped tests are working since that is a different issue.
The runner script does not currently spin up a database, since ideally these should be mocked for unit testing. A database instance may be required for integration testing but not unit testing.

## Developer Checklist
- [x] Make code change
- [x] Update tests
  - [x] Update / create new tests
  - [x] Ensure these tests have the expected behaviour
  - [x] Test locally and ensure tests are passing
- [x] Update documentation
  - [x] Readme
  - [x] Docstrings
  - [x] Comments
  - [x] Wiki

## Reviewer Checklist
- [ ] Check new code for code smells
- [ ] Check new tests
  - [ ] Ensure adequate coverage
  - [ ] Check for code smells within tests
- [ ] Check if documentation needs updating
  - [ ] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

### CODE REVIEW

**Comment by LukeParky** [comment]
Hi @caseyli92, since we were talking about testing earlier today this could be a good opportunity for you to review some test runner configuration.

Check the checklist above and also check the runner actions to make sure the tests are actually running.
This PR will fix the error in all the other PRs for unit tests

File: README.md [hunk 1]
```diff
@@ -27,11 +27,10 @@ conda env create -f environment.yml
 ```
 
 ## Tests
-Tests exist in the tests folder.
+Tests exist in the `tests/` folder.
```
Comment (caseyli92) [inline README.md:30]:
```suggestion
Tests exist in the `tests` folder.
```

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -20,37 +20,57 @@ jobs:
           flake8 src --show-source --count --statistics --max-line-length=120
 
   unit-tests:
-    # Starts the database, builds the testing docker image and runs all unit tests using docker
+    name: Run unit tests with pytest
```
Comment (caseyli92) [inline .github/workflows/run-tests.yml:23]:
```suggestion
    # Run unit tests with pytest
```

File: .github/workflows/run-tests.yml [hunk 3]
```diff
@@ -20,37 +20,57 @@ jobs:
           flake8 src --show-source --count --statistics --max-line-length=120
```
Comment (caseyli92) [inline .github/workflows/run-tests.yml:21]: add names to everything

File: .github/workflows/run-tests.yml [hunk 4]
```diff
@@ -20,37 +20,57 @@ jobs:
...
           echo import .env file from secrets and regenerate the file
           echo $ENV_BASE64 | base64 -d > .env
-      - name: Run unit tests
+
+      - name: Run tests with pytest
         run: |
-          docker run --network="host" unit_test
+          pytest
 
   sonarcloud:
```
Comment (caseyli92) [inline .github/workflows/run-tests.yml:75]: add names to everything

File: .github/workflows/run-tests.yml [hunk 5]
```diff
@@ -20,37 +20,57 @@ jobs:
...
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: digitaltwin
+          environment-file: environment.yml
+          auto-activate-base: false
+          channels: conda-forge
+          channel-priority: strict
+          use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
+          python-version: ${{ matrix.python-version }}
+          mamba-version: "*"
```
Comment (caseyli92) [inline .github/workflows/run-tests.yml:57]: as discussed, let's try running without mamba.

**Review by caseyli92 — CHANGES_REQUESTED** [review]
Looks good, just a couple of changes to be made. Great learning for me as well :) Thanks Luke!

**Review by caseyli92 — APPROVED** [review]
Excellent work :)

**Comment by LukeParky** [comment]
Closing PR to reopen so that checks are updated, Checks has a problem where linting-flake8 no longer exists but is required.

**Comment by LukeParky** [comment]
Merging without linting-flake8 since I can not disable the requirement due to github bug.

**Pull request merged**